training/setup/setup_functions.py

```diff
@@ -0,0 +1,356 @@
+            inter_list.sort(key=lambda x: x[1].casefold())
+            for list_index, list_val in enumerate(inter_list, start=1):
+                print("{:2d}. Instance Name: {}   |  "
+                      "Instance Location: {}  | "
+                      "Instance Plan: {} ".
+                      format(list_index, list_val[1], list_val[2],
+                             list_val[3]))
+                existing_instances.append(list_val[1])
+                existing_guids.append(list_val[4])
+            # Adding create new instance option.
+            if len(existing_instances) > 0:
+                print('{:2d}. {}'.format(int(count) + 1,
+                                         '* Create New Instance *'))
+                existing_instances.append('Create New Instance')
+                existing_guids.append('Create New Guid')
+            else:
+                # Default instance create option when no instances are found
+                print('{:2d}. {}'.format(1, 'Create New Instance'))
+            # Prompt user to input choice and return lists and choice.
+            if len(existing_instances) > 0 and \
+                    existing_instances[0] != 'Create New Instance':
+                while True:
+                    instance_option = input("[PROMPT] Your selection:  ") \
+                                      .strip()
+                    if not instance_option.isdigit() or \
+                       int(instance_option) < 1 or \
+                       (int(instance_option) >= (len(existing_instances) + 1)):
+                        print("[MESSAGE] Enter a number between 1 and "
+                              "{}.".format(len(existing_instances)))
+                        continue
+                    else:
+                        return existing_instances, instance_option, \
+                               existing_guids
+            else:
+                return ['Create New Instance'], '1', ['Create New Guid']
+
+        else:
+            print("[DEBUG] Failing with status code:", response.status_code)
+            print("[DEBUG] Reason for failure:", response.reason)
+            sys.exit()
+
+    def wml_key_details(self, available_keys, count,
+                        inter_wml_list):
+        """
+        Get details of the WML keys matching guid of the instance.
+        :param available_keys: response from the request
+        :param count: count to track number of retrieved keys
+                      under the selected instance.
+        :param inter_wml_list: intermediate list holding details of
+                               the key retrieved.
+        :return: intermediate list and count
+        """
+        for index, key in enumerate(available_keys['resources'], start=1):
+            count += 1
+            inter_wml_list.append([count, key['name'], key['guid']])
+
+        return inter_wml_list, count
+
+    def wml_key_check(self, instance_guid): # noqa
```

Comment:
The `# noqa`'s in this file don't seem necessary

Reply:
Thanks for the feedback, but I can't really change it here.. The setup scripts are managed here:
https://github.ibm.com/CODAIT/max-model-training/

Reply:
Tracking all changes related to training framework separately. Will make all in one.

Reply:
Thanks :)))